### PR TITLE
[Refactor] Migrate dependencies in docs/requirements.txt to pyproject.toml

### LINKED ISSUE #321: [Refactor] Migrate dependencies in docs/requirements.txt to pyproject.toml
As part of the migration to poetry for dependency management in pycytominer, we should migrate the dependencies listed in `docs/requirements.txt` to `pyproject.toml`. This includes the following tasks:

- [ ] For each dependency, run a command similar to `poetry add sphinx@~7.1 --group docs`
- [ ] Delete the `docs/requirements.txt` file
- [ ] Delete the lines in `readthedocs.yml` that install `docs/requirements.txt`
- [ ] Update the `poetry.lock` file

### PR DESCRIPTION
**Pull request by kenibrewer**
# Description

This migrates the dependencies for the docs build process into the pyproject.toml file to unify all dependency installs using Poetry. It also adds a docs preview pipeline that adds a preview link to any newly created PRs.

Fixes #321

## What is the nature of your change?

- [x] Bug fix (fixes an issue).

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

**Comment by kenibrewer** [comment]
@d33bs Is there a way to test the docs build process for this pull request? I'm not sure where to look for that.

**Comment by d33bs** [comment]
Hi @kenibrewer, previously I created a standalone Read the Docs deployment for testing via a fork here: https://readthedocs.org/dashboard/pycytominer-test-docs . I'd be glad to invite you to administrate this and use for testing if you'd like. Generally, I'd propose we move into a formalized testing process for documentation to cover scenarios like this one and others as they come up. Would you have any thoughts on the approach here?

**Comment by kenibrewer** [comment]
@d33bs I think we should configure the Pull Requests features for Read the Docs here described [here](https://docs.readthedocs.io/en/latest/pull-requests.html). That should allow us to use just one ReadTheDocs deployment instead of two. 

With that PR setting turned on, we should be able to use the Github action [here](https://github.com/readthedocs/actions/tree/v1/preview), to attach previews to any PR that affects docs.

**Review by d33bs — APPROVED** [review]
Great stuff with this PR, thank you for sending this through! I left a minor comment about naming for Github Actions, otherwise I think things are looking good.

Regarding your previous thoughts about the approach, I love the idea! Would you be interested in helping to enable the functionality on Readthedocs via the proper access there? Please let me know and I'll add you so you may configure things appropriately. If not, I can help out with the adjustments which are needed.

Either way, once we enable the functionality, do we need to do anything special to see the results in this PR?

**Comment by kenibrewer** [comment]
> Regarding your previous thoughts about the approach, I love the idea! Would you be interested in helping to enable the functionality on Readthedocs via the proper access there? Please let me know and I'll add you so you may configure things appropriately. If not, I can help out with the adjustments which are needed.

Sure! I'd be happy to figure that integration on ReadTheDocs. I think there might be an additional solution I might be able to enable if I'm a maintainer for both the Read The Docs project and Github using the ReadTheDocs Github app. That option should show the actual ReadTheDocs build process along with all the other pipelines defined with Github Actions. 

> Either way, once we enable the functionality, do we need to do anything special to see the results in this PR?

As currently set up, the workflow only runs when the pull request is created which is what is suggested in the ReadTheDocs Action documentation. I [tried](https://github.com/cytomining/pycytominer/actions/runs/6408306782/job/17396984689) enabling it to run on any pull request event, but I interpreted "Resource Not Available" error to mean that the event is designed only to run on PR creation because that's when it has permissions to edit the PR description. 

One corner case I'm aware of, is that if someone creates a PR without docs changes and then adds them later, this preview pipeline will never run. We could enable the pipeline to run on all PRs instead by getting rid of the paths filter. 

In the end though, the docs preview pipeline is a (useful) convenience function that will help contributors know where they can preview any changes they make to the docs. I think the full ReadTheDocs Github App integration is the workflow that will prevent us from merging any changes that break the docs build process.

File: .github/workflows/docs.yml
[the remark this answers is not in the document]
Comment (kenibrewer) [reply .github/workflows/docs.yml:1]: Done

**Pull request merged**